### ISSUE TITLE
Upgrade from `C118` -> `C119` on macOS x64 via 1.61.x - Nightly (Checking C119)

### ISSUE THREAD

**Issue by kjozwiak** [brave/brave-browser#33769]
As per process, QA runs through the following cases to ensure that the major chromium bump that's about to get merged into `master` doesn't regress the `Nightly` channel. Once the major chromium bump is merged into `master`, QA will run through a full manual pass on `Windows` & `Android`.

### Startup & Components

- [x] Verify that Brave is only contacting `*.brave.com` endpoints on first launch using either `Charles Proxy`, `Fiddler`, `Wireshark` or `LittleSnitch` (or a similar application)
  - [x] Verify that opening a NTP doesn't trigger any outbound connections related to widgets without user interaction
- [x] Remove the following component folders and ensure that they're being re-downloaded after restarting the browser:
  - [x] `afalakplffnnnlkncjhbmahjfjhmlkal`: `AutoplayWhitelist.dat`, `ExtensionWhitelist.dat`, `ReferrerWhitelist.json` and `Greaselion.json`
  - [x] `CertificateRevocation`
  - [x] `gccbbckogglekeggclmmekihdgdpdgoe`: (Sponsored New Tab Images)
  - [x] `gkboaolpopklhgplhaaiboijnklogmbc`: Brave Ad Block List Catalog
  - [x] `iodkpdagapdfkphljnddpjlldadblomo`: Brave Ad Block Updater
  - [x] `mfddibmblmbccpadfndgakiopmmhebop`: Brave Ad Block List Catalog
  - [x] `oofiananboodjbbmdelgdommihjbkfag`: HTTPSE
  - [x] `Safe Browsing`
- [x] Restart the browser, load `brave://components`, wait for 8 mins and verify that no component shows any errors

**Note:** Always double check `brave://components` to make sure there's no errors/missing version numbers

### Upgrade

- [x] Make sure that data from the last version appears in the new version OK
- [x] Ensure that `brave://version` lists the expected Brave & Chromium versions
- [x] With data from the last version, verify that:
  - [x] Bookmarks on the bookmark toolbar and bookmark folders can be opened
  - [x] Cookies are preserved
  - [x] Installed extensions are retained and work correctly
  - [x] Opened tabs can be reloaded
  - [x] Stored passwords are preserved
  - [x] Sync chain created in previous version is retained
  - [x] Social media-blocking buttons changes are retained
  - [x] Custom filters under brave://settings/shields/filters are retained
  - [x] Custom lists under brave://settings/shields/filters are retained
  - [x] Rewards
    - [x] BAT balance is retained
    - [x] Auto-contribute list is retained
    - [x] Both Tips and Monthly Contributions are retained
    - [x] Panel transactions list is retained
    - [x] Changes to rewards settings are retained
    - [x] Ensure that Auto Contribute is not being enabled when upgrading to a new version if AC was disabled
  - [x] Ads
    - [x] Both `Estimated pending rewards` & `Ad notifications received this month` are retained
    - [x] Changes to ads settings are retained
    - [x] Ensure that ads are not being enabled when upgrading to a new version if they were disabled
    - [x] Ensure that ads are not disabled when upgrading to a new version if they were enabled

**Comment by kjozwiak** [comment]
Verification **PASSED** on `macOS 14.0 x64 Sonoma` using the following build(s):

```
Brave | 1.61.30 Chromium: 118.0.5993.70 (Official Build) nightly (x86_64)
--- | ---
Revision | 7e4b32142a514fc5de17047cb19c5b6c06c00105
OS | macOS Version 14.0 (Build 23A344)
```

**`Prerequisite:`** Ensure that you're connect to Brave's VPN to download/use the staging server as per @mherrmann.

<details>
<summary>C119 --> C119 on-demand upgrade check - PASSED</summary>

1. Install `C119` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1697767820471179?thread_ts=1697655851.387759&cid=CU5UXM3EC
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/cr-bump-update-tests-x64/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```
[6233:259:1022/230148.294572:INFO:sparkle_glue.mm(323)] brave update: did finish loading appcast
[6233:259:1022/230148.295100:INFO:sparkle_glue.mm(329)] brave update: did find valid update with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
[6233:259:1022/230148.296400:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
[6233:259:1022/230158.164499:INFO:sparkle_glue.mm(386)] brave update: will install update on quit with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
```

7. Confirm you are prompted to "Relaunch" to get update

![Screenshot 2023-10-22 at 11 02 29 PM](https://github.com/brave/brave-browser/assets/2602313/0de32717-d43c-460b-9c31-c64c4aff6e32)

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------
![Screenshot 2023-10-22 at 11 02 57 PM](https://github.com/brave/brave-browser/assets/2602313/771a24db-c010-44ce-8483-837a3598bdff) | ![Screenshot 2023-10-22 at 11 02 49 PM](https://github.com/brave/brave-browser/assets/2602313/70db3e20-b51a-4a68-9c6c-74485a7d9bd7)

</details>


<details>
<summary>C118 --> C119 on-demand upgrade check - PASSED</summary>

1. Install `C117` build via https://github.com/brave/brave-browser/releases/tag/v1.61.30
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/cr-bump-update-tests-x64/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```
[6541:259:1022/232105.040064:INFO:sparkle_glue.mm(323)] brave update: did finish loading appcast
[6541:259:1022/232105.040569:INFO:sparkle_glue.mm(329)] brave update: did find valid update with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
[6541:259:1022/232105.041828:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
[6541:259:1022/232229.389841:INFO:sparkle_glue.mm(386)] brave update: will install update on quit with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
```

7. Confirm you are prompted to "Relaunch" to get update

![Screenshot 2023-10-22 at 11 22 41 PM](https://github.com/brave/brave-browser/assets/2602313/5d2387c4-68b9-4206-90e6-6259a214f60d)

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------
![Screenshot 2023-10-22 at 11 23 52 PM](https://github.com/brave/brave-browser/assets/2602313/5e0438ae-93a1-49f5-bf97-2f37a56f1a00) | ![Screenshot 2023-10-22 at 11 24 00 PM](https://github.com/brave/brave-browser/assets/2602313/a2761a9b-b2d7-4911-93dc-6c4f9b0dbade)

</details>

<details>
<summary>C119 --> C119 background upgrade check - PASSED</summary>

1. Install `C119` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1697767820471179?thread_ts=1697655851.387759&cid=CU5UXM3EC
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/cr-bump-update-tests-x64/appcast.xml`
8. DO NOT navigate to `brave://settings/help`
9. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```
[950:259:1022/233658.696728:INFO:sparkle_glue.mm(323)] brave update: did finish loading appcast
[950:259:1022/233658.697673:INFO:sparkle_glue.mm(329)] brave update: did find valid update with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
[950:259:1022/233658.698147:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
[950:259:1022/233709.305821:INFO:sparkle_glue.mm(386)] brave update: will install update on quit with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
```

10. Close and restart the browser via terminal with same command as step 7
11. Check version is updated under `brave://version`

![Screenshot 2023-10-22 at 11 38 02 PM](https://github.com/brave/brave-browser/assets/2602313/fe9ee8c2-9518-436d-a921-4a575f9d1be1)

</details>

<details>
<summary>C118 --> C119 background upgrade check - PASSED</summary>

1. Install `C117` build via https://github.com/brave/brave-browser/releases/tag/v1.61.30
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/cr-bump-update-tests-x64/appcast.xml`
8. DO NOT navigate to `brave://settings/help`
9. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```
[868:259:1022/234503.347795:INFO:sparkle_glue.mm(323)] brave update: did finish loading appcast
[868:259:1022/234503.348448:INFO:sparkle_glue.mm(329)] brave update: did find valid update with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
[868:259:1022/234503.348885:INFO:sparkle_glue.mm(350)] brave update: willDownloadUpdate with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
[868:259:1022/234550.522837:INFO:sparkle_glue.mm(386)] brave update: will install update on quit with AppcastItem(Date: Fri, 20 Oct 2023 20:31:02 +0000, Version: 1.61.39.0)
```

10. Close and restart the browser via terminal with same command as step 7
11. Check version is updated under `brave://version`

![Screenshot 2023-10-22 at 11 47 36 PM](https://github.com/brave/brave-browser/assets/2602313/a458e692-ab58-4ec4-b431-1a50e9987831)
</details>

**Comment by stephendonner** [comment]
Verification **PASSED** on `macOS 14.0 x64 Sonoma` using the following build(s):

```

```

**`Prerequisite:`** Ensure that you're connect to Brave's VPN to download/use the staging server as per @mherrmann.

<details>
<summary>C119 --> C119 on-demand upgrade check - PASSED</summary>

1. Install `C119` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1697767820471179?thread_ts=1697655851.387759&cid=CU5UXM3EC
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/cr-bump-update-tests-x64/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```

```

7. Confirm you are prompted to "Relaunch" to get update



Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------


</details>


<details>
<summary>C118 --> C119 on-demand upgrade check - PENDING</summary>

1. Install `C117` build via https://github.com/brave/brave-browser/releases/tag/v1.61.30
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but close this tab prior to closing the browser)
4. Close `Brave`
5. Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/cr-bump-update-tests-x64/appcast.xml`
6. Navigate to `brave://settings/help` and ensured that an update has been found/being applied

```

```

7. Confirm you are prompted to "Relaunch" to get update

![Screenshot 2023-10-22 at 11 22 41 PM](https://github.com/brave/brave-browser/assets/2602313/5d2387c4-68b9-4206-90e6-6259a214f60d)

Once relaunched, ensured the correct version was installed as per the following:

**`Example`** | **`Example`**
----------------|----------------


</details>

<details>
<summary>C119 --> C119 background upgrade check - PENDING</summary>

1. Install `C119` PR build (x64 builds available via https://bravesoftware.slack.com/archives/CU5UXM3EC/p1697767820471179?thread_ts=1697655851.387759&cid=CU5UXM3EC
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/cr-bump-update-tests-x64/appcast.xml`
8. DO NOT navigate to `brave://settings/help`
9. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```

```

10. Close and restart the browser via terminal with same command as step 7
11. Check version is updated under `brave://version`


</details>

<details>
<summary>C118 --> C119 background upgrade check - PENDING</summary>

1. Install `C117` build via https://github.com/brave/brave-browser/releases/tag/v1.61.30
2. Launch via terminal as follows: `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/nothanks/appcast.xml` (this is so you don't pick up any newer Nightly versions that have already been uploaded to sparkle)
3. Watch logs, confirm no update found (can also visit `brave://settings/help` but be sure to close this tab prior to closing the browser)
4. Close browser
5. Delete the following:
- `com.brave.Browser.nightly` folder from `~/Library/Caches`
- `com.brave.Browser.nightly.plist` file from `~/Library/Preferences`
- Alternatively you can probably advance your computer clock ~4 hours, I am fairly certain deleting at least one of the above files removes the timer check for background updates.
6. Restart your computer if you deleted files (if you just advanced computer clock this step isn't necessary)
7. After computer restart, Relaunch with `/Applications/Brave\ Browser\ Nightly.app/Contents/MacOS/Brave\ Browser\ Nightly --args --enable-logging=stderr --update-feed-url=https://updates.brave.com/sparkle/Brave-Browser/cr-bump-update-tests-x64/appcast.xml`
8. DO NOT navigate to `brave://settings/help`
9. Watch the logs and see that an update is found (be sure you have all of the below before proceeding)

```

```

10. Close and restart the browser via terminal with same command as step 7
11. Check version is updated under `brave://version`


</details>
